dubbo-registry/dubbo-registry-api/src/test/java/org/apache/dubbo/registry/client/metadata/ServiceInstanceMetadataUtilsTest.java

```diff
@@ -1,88 +0,0 @@
-
-    private static URL url = URL.valueOf("dubbo://192.168.0.102:20880/org.apache.dubbo.metadata.MetadataService?&anyhost=true&application=spring-cloud-alibaba-dubbo-provider&bind.ip=192.168.0.102&bind.port=20880&default.deprecated=false&default.dynamic=false&default.register=true&deprecated=false&dubbo=2.0.2&dynamic=false&generic=false&group=spring-cloud-alibaba-dubbo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=getAllServiceKeys,getServiceRestMetadata,getExportedURLs,getAllExportedURLs&pid=58350&register=true&release=2.7.1&revision=1.0.0&side=provider&timestamp=1557928573174&version=1.0.0");
-    private static URL url2 = URL.valueOf("rest://192.168.0.102:20880/org.apache.dubbo.metadata.MetadataService?&anyhost=true&application=spring-cloud-alibaba-dubbo-provider&bind.ip=192.168.0.102&bind.port=20880&default.deprecated=false&default.dynamic=false&default.register=true&deprecated=false&dubbo=2.0.2&dynamic=false&generic=false&group=spring-cloud-alibaba-dubbo-provider&interface=org.apache.dubbo.metadata.MetadataService&methods=getAllServiceKeys,getServiceRestMetadata,getExportedURLs,getAllExportedURLs&pid=58350&register=true&release=2.7.1&revision=1.0.0&side=provider&timestamp=1557928573174&version=1.0.0");
-
-    private static final String VALUE = "{\"rest\":{\"version\":\"1.0.0\",\"dubbo\":\"2.0.2\",\"release\":\"2.7.1\",\"port\":\"20880\"},\"dubbo\":{\"version\":\"1.0.0\",\"dubbo\":\"2.0.2\",\"release\":\"2.7.1\",\"port\":\"20880\"}}";
-    @Test
-    public void testMetadataServiceURLParameters() {
-
-        List<URL> urls = Arrays.asList(url, url2);
-
-        String parameter = ServiceInstanceMetadataUtils.getMetadataServiceParameter(urls);
-
-        JSONObject jsonObject = JSON.parseObject(parameter);
-
-        urls.forEach(url -> {
-            JSONObject map = jsonObject.getJSONObject(url.getProtocol());
-            for (Map.Entry<String, String> param : url.getParameters().entrySet()) {
-                String value = map.getString(param.getKey());
-                if (value != null) {
-                    assertEquals(param.getValue(), value);
-                }
-            }
-        });
-
-        assertEquals(VALUE, parameter);
-    }
-
-    @Test
-    public void testProtocolPorts() {
-
-//        Map<String, String> metadata = new LinkedHashMap<>();
-//
-//        String key = protocolPortMetadataKey("dubbo");
-//        assertEquals("dubbo.protocols.dubbo.port", key);
-//
-//        metadata.put(key, "20880");
-//
-//        key = protocolPortMetadataKey("rest");
-//        assertEquals("dubbo.protocols.rest.port", key);
-//
-//        metadata.put(key, "8080");
-//
-//        Map<String, Integer> protocolPorts = getProtocolPorts(metadata);
-//
-//        Map<String, Integer> expected = new LinkedHashMap<>();
-//
-//        expected.put("dubbo", 20880);
-//        expected.put("rest", 8080);
-//
-//        assertEquals(expected, protocolPorts);
-    }
-}
```

Comment:
could you pls describe why you remove this file?

Reply:
Because ` ServiceInstanceMetadataUtils. GetMetadataServiceParameter () ` the underlying implementation is gson.toJson(), there is no need to use outside gson API to test. (The original implementation was fastjson)